frontend/src/CreateGame.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { fetchAPI, postAPI } from './Api';
+import DatePicker from 'react-datepicker'
+
+interface IState {
+    roles: Map<string, boolean>
+    date: Date
+}
+
+interface IProps {
+}
+
+export class CreateGame extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props)
+        this.state = {roles: new Map<string, boolean>(), date: new Date()}
+    }
+
+    async fetchRoles() {
+        const roles = new Map<string, boolean>()
+        await fetchAPI('/game/roles')
+            .then(ls => ls.forEach((el: string) => { roles.set(el, false)}))
+        this.setState({ roles: roles })
+    }
+
+    async pushGame() {
+        const date = this.state.date
+        const roles = this.state.roles
+        await postAPI('/game', {
+                date: date, 
+                roles: [...roles.keys()].filter(el => true ==  roles.get(el))
+              }).then(data => { console.log(data)})
```

Comment:
You can use either `await` or `.then()` since they do the same thing.

Since we use `await` in other places in the code, it is best to use that here as well.
```suggestion
        const data = await postAPI('/game', {
                date: date, 
                roles: [...roles.keys()].filter(el => true ==  roles.get(el))
        });
        console.log(data);
              
```